android/src/main/java/com/reactnativecommunity/webview/RNCWebViewManager.java

```diff
@@ -749,17 +749,7 @@ public void receiveCommand(@NonNull WebView root, String commandId, @Nullable Re
           RNCWebView reactWebView = (RNCWebView) root;
           JSONObject eventInitDict = new JSONObject();
           eventInitDict.put("data", args.getString(0));
-          reactWebView.evaluateJavascriptWithFallback("(function () {" +
-            "var event;" +
-            "var data = " + eventInitDict.toString() + ";" +
-            "try {" +
-            "event = new MessageEvent('message', data);" +
-            "} catch (e) {" +
-            "event = document.createEvent('MessageEvent');" +
-            "event.initMessageEvent('message', true, true, data.data, data.origin, data.lastEventId, data.source);" +
-            "}" +
-            "document.dispatchEvent(event);" +
-            "})();");
+          reactWebView.evaluateJavascriptWithFallback("document.dispatchEvent(new MessageEvent('message', " + eventInitDict.toString() + "));");
```

Comment:
While we are here: this is not a correct way to encode json objects in js context
Please use the logic from https://github.com/ExodusMovement/schemasafe/blob/master/src/safe-format.js#L21-L34

Reply:
@ChALkeR I already thought about this issue. However, right now `eventInitDict` will always be of the format `{"data": "SOME STRING"}` (it's created two lines above), so it shouldn't be possible to use `__proto__` in this context. But I agree that this behavior could be dangerous if the code changes in the future.

>  Please use the logic from https://github.com/ExodusMovement/schemasafe/blob/master/src/safe-format.js#L21-L34

Since we don't need to encode a JSON object with an arbitrary shape, but just a wrapped string, we could also just base64 encode it. That should be easier to implement and it'd be more obvious that there's no possible injection point.

Reply:
Correct, it's worth noting that the string contents are however a JSON-encoded payload.

Perhaps we can investigate removing these functions completely and integrate `@exodus/json-rpc` (when open sourced) & `@exodus/schemasafe` soon.